sql/src/main/java/io/crate/operation/collect/sources/ShardCollectSource.java

```diff
@@ -288,20 +302,18 @@ private CrateCollector createMultiShardScoreDocCollector(RoutedCollectPhase coll
                 try {
                     shardInjector = indexService.shardInjectorSafe(shardId);
                     ShardCollectService shardCollectService = shardInjector.getInstance(ShardCollectService.class);
-                    CrateCollector collector = shardCollectService.getDocCollector(
+                    CrateCollector.Builder collector = shardCollectService.getCollectorBuilder(
                         collectPhase,
-                        projectorChain,
+                        downstreamRequirements,
                         jobCollectContext
                     );
                     crateCollectors.add(collector);
                 } catch (ShardNotFoundException | IllegalIndexShardStateException e) {
                     crateCollectors.add(remoteCollectorFactory.createCollector(
-                        indexName, shardId, collectPhase, projectorChain, jobCollectContext.queryPhaseRamAccountingContext()));
+                        indexName, shardId, collectPhase, jobCollectContext.queryPhaseRamAccountingContext()));
                 } catch (InterruptedException e) {
-                    projectorChain.fail(e);
```

Comment:
I think this was necessary because `newShardDownstreamProjector` of the shardChain incremented counters on the RowMerger. So the same number of failure calls had to be made.

Since `getCollectorBuilder` doesn't cause those `newShardDownstreamProjector` calls anymore this shouldn't be necessary anymore.